src/web/routes/entities/index.ts

```diff
@@ -2,47 +2,48 @@ import { HttpStatusCode } from 'axios';
 import { Router } from 'express';
 import type { NextFunction } from 'express';
 
+import {
+	ASSOCIATE_ENTITY_REQUEST_BODY_SCHEMA,
+	DISASSOCIATE_ENTITY_REQUEST_BODY_SCHEMA,
+} from './schemas';
 import type {
-	AssociateEntityRequestParams,
+	AssociateEntityRequest,
 	AssociateEntityResponse,
-	DisassociateEntityRequestParams,
+	DisassociateEntityRequest,
 	DisassociateEntityResponse,
 } from './types';
 
+import { assertSchema } from '../../../infrastructure';
 import {
 	associateEntityUseCase,
 	disassociateEntityUseCase,
 } from '../../../usecases';
-import { authHeaderSymmetricJwtMiddleware } from '../../middleware';
-import type { TypedRequest } from '../types';
-
-export class UnauthorizedError extends Error {}
-
-const getUserIdHeaderOrThrow = (
-	userId: string | string[] | undefined,
-): string => {
-	if (!userId || typeof userId !== 'string') {
-		throw new UnauthorizedError('Missing or invalid User-Id header');
-	}
-	return userId;
-};
+import {
+	authHeaderSymmetricJwtMiddleware,
+	extractUserIdFromHeadersMiddleware,
+} from '../../middleware';
 
 export const entitiesRouter = Router();
 
+entitiesRouter.use(
+	extractUserIdFromHeadersMiddleware,
+	authHeaderSymmetricJwtMiddleware,
+);
+
 entitiesRouter.post(
 	'/associateEntity',
-	authHeaderSymmetricJwtMiddleware,
 	(
-		req: TypedRequest<AssociateEntityRequestParams>,
+		req: AssociateEntityRequest,
 		res: AssociateEntityResponse,
 		next: NextFunction,
 	) => {
-		const atlassianUserId = getUserIdHeaderOrThrow(req.headers['user-id']);
+		assertSchema(req.body, ASSOCIATE_ENTITY_REQUEST_BODY_SCHEMA);
```

Comment:
**praise:** Amazing! I feel much more confident in what our service does with this assertion.